modules/build/src/main/scala/scala/build/options/validation/BuildOptionsRule.scala

```diff
@@ -0,0 +1,34 @@
+package scala.build.options.validation
+
+import scala.build.Position
+import scala.build.errors.BuildException
+import scala.build.options.BuildOptions
+
+trait BuildOptionsRule {
+  def validate(options: BuildOptions): Either[BuildException, Unit]
+}
+
+object BuildOptionsRule {
+  def validateAll(options: BuildOptions): Either[BuildException, Unit] =
+    List(JvmOptionsForNonJvmBuild).map(_.validate(options)).partition(_.isLeft) match {
+      case (Nil, _)  => Right(())
+      case (some, _) => some.head
+    }
+}
+
+class ValidationException(
+  message: String,
+  positions: Seq[Position] = Nil
+) extends BuildException(message, positions)
+
+object JvmOptionsForNonJvmBuild extends BuildOptionsRule {
+  def validate(options: BuildOptions): Either[BuildException, Unit] = {
+    val jvmOptions = options.javaOptions.javaOpts.find(p => p.value.nonEmpty)
+    if (jvmOptions.nonEmpty && options.platform.value != scala.build.options.Platform.JVM)
```

Comment:
As I think I heard someone discuss that earlier today, this might be a problem if users just want to give native a try by passing `--native` on the command-line (while some of their sources have using directives adding Java options).

So we may want to not raise an error if the platform originates from the command-line (which can be checked via its position) while the Java options don't.

Reply:
I agree. Should we report a warning then, or just continue silently?

Reply:
Ok, If I understand the next comment correctly, you would prefer a warning. :+1:

Reply:
Yeah, but we don't have a "channel" to report warnings yet (I mean, a way to pass around `BuildException`s that should be considered warnings, up to the user). So no warnings nor errors would be fine for now IMO.

Reply:
It was printing a warning, not an error. I prepared an ad hoc solution to handle warnings. I added a check in the test to ensure the program is not failing and the message is reported with [warn]